src/compiler/scala/tools/nsc/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -478,7 +478,7 @@ trait BCodeBodyBuilder extends BCodeSkelBuilder {
      * Otherwise it's safe to call from multiple threads.
      */
     def genConstant(const: Constant): Unit = {
-      (const.tag: @switch) match {
+      (const.tag/*: @switch*/) match {
```

Comment:
It would be interesting to know why Dotty cannot emit a switch here but scalac can

Reply:
My bad, scalac also emits the warning. However I don't see why neither scalac nor notty can emit a switch. I think the user would benefit from a bit more information in the warning